tests/test_lib_tokens_totp.py

```diff
@@ -727,6 +728,39 @@ def test_26_get_setting_type(self):
         r = TotpTokenClass.get_setting_type("totp.blabla")
         self.assertEqual(r, "")
 
+    def test_26_is_previous_otp(self):
+        # check if the OTP was used previously
+        serial = "previous"
+        db_token = Token(serial, tokentype="totp")
+        db_token.save()
+        token = TotpTokenClass(db_token)
+        token.set_otpkey(self.otpkey)
+        token.set_hashlib("sha1")
+        token.set_otplen(6)
+        # Authenticate with the current OTP value
+        counter = token._time2counter(time.time(), timeStepping=30)
+        otp_now = token._calc_otp(counter)
+        r = token.check_otp(otp_now, window=180)
+        self.assertEqual(r, counter)
+        # Now we try several is_previous_otp and the timeShift must stay the same!
+        ts0 = float(token.get_tokeninfo("timeShift"))
+        self.assertTrue(-31 < ts0 < 31)
```

Comment:
shouldn't this be somewhere between -180 and +180?

Reply:
I think in theory you are right. The user *could* wait more than 31 seconds or the clock could drift up to 3 minutes.
Well in this test case it would not happen, that we spend so much time between line #742 and #743.

Fixed in 11fce212